deepchem/feat/material_featurizers/lcnn_featurizer.py

```diff
@@ -0,0 +1,687 @@
+  determined by first extracting a site local environment from the primitive cell,
+  and perform graph matching and distance matching to find neighbors.
+  First, the template of the Primitive cell needs to be defined along with periodic
+  boundary conditions and active and specator site details. structure(Data Point
+  i.e different configuration of adsorbate atoms) is passed for featurization.
+
+  This particular featurisation produces a regular-graph (equal number of Neighbors)
+  along with its permutation in 6 symmetric axis. This transformation can be
+  applied when orderering of neighboring of nodes around a site play an important role
+  in the propert predictions. Due to consideration of local neighbor environment,
+  this current implementation would be fruitful in finding neighbors for calculating
+  formation energy of adbsorption tasks where the local. Adsorption turns out to be important
+  in many applications such as catalyst and semiconductor design.
+
+  The permuted neighbors are calculated using the Primitive cells i.e periodic cells
+  in all the data points are built via lattice transformation of the primitive cell.
+
+  `Primitive cell Format:`
+
+  1. Pymatgen structure object with site_properties key value
+   - "SiteTypes" mentioning if it is a active site "A1" or spectator
+     site "S1".
+  2. ns , the number of spectator types elements. For "S1" its 1.
+  3. na , the number of active types elements. For "A1" its 1.
+  4. aos, the different species of active elements "A1".
+  5. pbc, the periodic boundary conditions.
+
+  `Data point Structure Format(Configuration of Atoms):`
+
+  1. Pymatgen structure object with site_properties with following key value.
+   - "SiteTypes", mentioning if it is a active site "A1" or spectator
+     site "S1".
+   - "oss", different occupational sites. For spectator sites make it -1.
+
+  It is highly recommended that cells of data are directly redefined from
+  the primitive cell, specifically, the relative coordinates between sites
+  are consistent so that the lattice is non-deviated.
+
+  References
+  ----------
+  .. [1] Jonathan Lym and Geun Ho Gu, J. Phys. Chem. C 2019, 123, 18951−18959
+
+  Examples
+  --------
+  >>> PRIMITIVE_CELL = {
+  >>>   "lattice": [[2.818528, 0.0, 0.0],
+  >>>               [-1.409264, 2.440917, 0.0],
+  >>>               [0.0, 0.0, 25.508255]],
+  >>>   "coords": [[0.66667, 0.33333, 0.090221],
+  >>>              [0.33333, 0.66667, 0.18043936],
+  >>>              [0.0, 0.0, 0.27065772],
+  >>>              [0.66667, 0.33333, 0.36087608],
+  >>>              [0.33333, 0.66667, 0.45109444],
+  >>>              [0.0, 0.0, 0.49656991]],
+  >>>   "species": ['H', 'H', 'H', 'H', 'H', 'He'],
+  >>>   "site_properties": {'SiteTypes': ['S1', 'S1', 'S1', 'S1', 'S1', 'A1']}
+  >>> }
+  >>> PRIMITIVE_CELL_INF0 = {
+  >>>    "cutoff": np.around(6.00),
+  >>>    "structure": Structure(**PRIMITIVE_CELL),
```

Comment:
I think this will fail the doctest since we don't import `Structure` in this example